src/main/java/de/tum/in/www1/artemis/service/programming/RepositoryService.java

```diff
@@ -72,12 +86,54 @@ public Map<String, FileType> getFiles(Repository repository) {
     }
 
     /**
-     * Get all files/folders with content from repository.
+     * Retrieves the content of files at a specific commit in a given repository.
+     *
+     * @param programmingExercise The programming exercise which contains the VCS repository. This is used to determine the repository URI for TESTS repositories.
+     * @param commitId            The commit identifier from which to extract file contents.
+     * @param repositoryType      The type of the repository (e.g., TESTS, TEMPLATE, etc.). Relevant for participations of type TESTS.
+     * @param participation       The participation related to the repository.
+     * @return A map where each key is a file path and each value is the content of the file as a String. This represents the state of the repository at the given commit.
+     * @throws IOException     If an I/O error occurs during the file content retrieval process. This could be due to issues with file access, network problems, etc.
+     * @throws GitAPIException If an error occurs while interacting with the Git repository. This could be due to issues with repository access, invalid commit ids, etc.
+     */
+    public Map<String, String> getFilesContentAtCommit(ProgrammingExercise programmingExercise, String commitId, RepositoryType repositoryType,
+            ProgrammingExerciseParticipation participation) throws IOException, GitAPIException {
+        // Check if local VCS is active
+        if (profileService.isLocalVcsActive()) {
+            log.info("Using local VCS for getting files at commit {} for participation {}", commitId, participation.getId());
+            // If local VCS is active, operate directly on the bare repository
+            var repoUri = Objects.equals(repositoryType, RepositoryType.TESTS) ? programmingExercise.getVcsTestRepositoryUri() : participation.getVcsRepositoryUri();
+            Repository repository = gitService.getBareRepository(repoUri);
+            return getFilesContentFromBareRepository(repository, commitId);
+        }
+        else {
+            Repository repository;
+            // If the repository type is tests, check out the tests repository
```

Comment:
Does this comment really add anything? If yes, why is it not also present in the localVC case?

Reply:
I removed it